build-system/tasks/check-video-interface-list.js

```diff
@@ -0,0 +1,132 @@
+/**
+ * Copyright 2021 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+const argv = require('minimist')(process.argv.slice(2));
+const globby = require('globby');
+const tempy = require('tempy');
+const {blue, bold, cyan, red} = require('kleur/colors');
+const {getStdout} = require('../common/process');
+const {log, logWithoutTimestamp} = require('../common/logging');
+const {readFile, writeFile} = require('fs-extra');
+
+/** Checks or updates 3rd party video player list on this Markdown file. */
+const filepath = 'spec/amp-video-interface.md';
+
+/** Excludes these extensions since they're on a separate list. */
+const excludeGeneric = ['amp-video', 'amp-video-iframe'];
+
+/** Determines whether a file belongs to an extension that should be listed. */
+const grepJsContent = '@implements {.*VideoInterface}';
+
+/** Finds extension files here. */
+const grepJsFiles = 'extensions/**/*.js';
+
+/**
+ * Returns a formatted list entry.
+ * @param {string} name
+ * @return {name}
+ */
+const entry = (name) =>
+  `-   [${name}](https://amp.dev/documentation/components/${name})\n`;
+
+/**
+ * Generates Markdown list by finding matching extensions.
+ * @return {string}
+ */
+const generateList = () =>
+  getStdout(
+    ['grep -lr', `"${grepJsContent}"`, ...globby.sync(grepJsFiles)].join(' ')
+  )
+    .trim()
+    .split('\n')
+    .reduce((list, path) => {
+      const name = path.substr('extensions/'.length).split('/').shift();
+      return list + (excludeGeneric.includes(name) ? '' : entry(name));
+    }, '');
+
+/**
+ * Returns a RegExp that matches the existing list.
+ * @return {RegExp}
+ */
+const getListRegExp = () =>
+  new RegExp(
+    `(${entry('NAME')
+      .replace(/[.*+?^${}()|[\]\\]/g, '\\$&')
+      .replace(/\NAME/g, `((?!${excludeGeneric.join('|')})[a-z0-9-]+)`)})+`,
+    'gim'
+  );
+
+/**
+ * Diffs a file against content that might replace it.
+ * @param {string} filepath
+ * @param {string} content
+ * @return {!Promise<string>}
+ */
+const diffTentative = (filepath, content) =>
+  tempy.write.task(content, (temporary) =>
+    getStdout(
+      `git -c color.ui=always diff -U1 ${filepath} ${temporary}`
+    ).replace(new RegExp(temporary, 'g'), `/${filepath}`)
+  );
+
+/**
+ * @param {string} filepath
+ * @param {string} content
+ * @param {string} output
+ * @param {boolean} write
+ */
+async function writeOrFailWhenUnequal(
```

Comment:
@rsimha I will extract and share `writeOrFailWhenUnequal` in a later PR to accomplish the same for [the z-index list](https://github.com/ampproject/amphtml/blob/master/css/Z_INDEX.md). I was thinking somewhere under `tasks/helpers/` but I don't know how you feel about that directory existing alongside `tasks/helpers.js`

Reply:
Yeah, the helper function situation has gotten a little messy. The files `tasks/helpers.js` and `tasks/extension-helpers.js` are primarily used by our compilation pipeline, so I was planning on moving them to `compile/` at some point.

Maybe you could introduce a new file called `tasks/utils.js` for stuff that's common across our dev tasks? (This is the model used by all the stuff inside `pr-check/`.)

Reply:
I just realized we already have `build-system/common/`. I think that's a better place to stick this function.